iam.tf

```diff
@@ -36,3 +36,19 @@ resource "google_project_iam_member" "this" {
   role    = "roles/browser"
   member  = "serviceAccount:${google_service_account.cleaner.email}"
 }
+
+# Grant cleaner service account roles/artifactregistry.writer role to read the repository
+# and delete container images.
+resource "google_artifact_registry_repository_iam_member" "this" {
+  for_each = {
+    for repo in var.gar_repositories : repo.name => repo
+  }
+
+  project    = each.value.project_id != null ? each.value.project_id : local.google_project_id
+  location   = each.value.region
+  repository = "projects/${each.value.project_id != null ? each.value.project_id : local.google_project_id}/locations/${each.value.region}/repositories/${each.value.name}"
+  role       = "roles/artifactregistry.writer"
```

Comment:
```suggestion
  role       = "roles/artifactregistry.repoAdmin"
```

Reply:
It seems that the `writer` role is missing the `artifactregistry.repositories.deleteArtifacts` permission !

Reply:
see https://cloud.google.com/artifact-registry/docs/access-control#roles